### ISSUE TITLE
Ruff incorrectly shows UP013 violations for functional TypedDicts that cannot be converted to class syntax

### ISSUE THREAD

**Issue by sjdemartini** [charliermarsh/ruff#3147]
Ruff currently shows a violation for the pyupgrade rule UP013 "UP013 Convert X from TypedDict functional to class syntax", even if the TypedDict `X` cannot be converted to class syntax since it has keys that aren't valid identifiers.

Example code that produces this violation:

```python
from typing import TypedDict
Point2D = TypedDict('Point2D', {'in': int, 'x-y': int})
```

Ruff version: 0.0.251

---

**More details**: 

Per the python official docs [here](https://docs.python.org/3/library/typing.html#typing.TypedDict):

> The functional syntax should also be used when any of the keys are not valid [identifiers](https://docs.python.org/3/reference/lexical_analysis.html#identifiers), for example because they are keywords or contain hyphens. Example:
>    ```
>    # raises SyntaxError
>    class Point2D(TypedDict):
>        in: int  # 'in' is a keyword
>        x-y: int  # name with hyphens
>    
>    # OK, functional syntax
>    Point2D = TypedDict('Point2D', {'in': int, 'x-y': int})
>    ```

As such, Ruff's current behavior of showing a violation of "UP013 Convert Point2D from TypedDict functional to class syntax" in the above functional syntax example seems undesirable, since it's actually the recommended/only way to use TypedDict. Rather, Ruff should just silently ignore. (pyupgrade correctly ignores this scenario.) 

Originally discussed here https://github.com/charliermarsh/ruff/issues/1212#issuecomment-1440902486
